basyx.examples.timerseries/src/main/java/org/eclipse/basyx/examples/timeseries/TimeSeriesSubmodel/TimeSeriesSubmodel.java

```diff
@@ -0,0 +1,127 @@
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * 
+ * SPDX-License-Identifier: MIT
+ ******************************************************************************/
+
+package org.eclipse.basyx.examples.timeseries.TimeSeriesSubmodel;
+
+import java.util.function.Function;
+
+import org.eclipse.basyx.aas.bundle.AASBundle;
+import org.eclipse.basyx.submodel.metamodel.api.ISubmodel;
+import org.eclipse.basyx.submodel.metamodel.map.Submodel;
+import org.eclipse.basyx.submodel.metamodel.map.submodelelement.SubmodelElementCollection;
+import org.eclipse.basyx.submodel.metamodel.map.submodelelement.dataelement.MultiLanguageProperty;
+import org.eclipse.basyx.submodel.metamodel.map.submodelelement.operation.Operation;
+
+import com.influxdb.annotations.Measurement;
+
+import org.eclipse.basyx.examples.timeseries.AASXImporter;
+
+/*
+ * Creates ReadRecords operation for a TimeSeries Submodel that follows the IDTA standard
+ * (https://github.com/admin-shell-io/submodel-templates). The operation takes a Range SubmodelElement as input, which
+ * is timerange in ISO UTC format. The output is a SubmodelElementCollection consisting of records
+ * (https://admin-shell.io/idta/TimeSeries/Record/1/1) as found under TimeSeries->Metadata->Record.
+ * Creates another experimental operation ReadRecordsShort which has a similar input as ReadRecords, but
+ * returns the query results as a string in a value of a property (reduces overhead).
+ * 
+ * author Al-Obaidi
+ */
+public class TimeSeriesSubmodel {
+
+    private static final String TIMESERIES_SUBMODEL_IDSHORT = "TimeSeries";
+    private static final String METADATA_SME_IDSHORT = "Metadata";
+    private static final String RECORD_SME_IDSHORT = "Record";
+    private static final String NAME_SME_IDSHORT = "Name";
+    private static final String READRECORDS_OPERATION_IDSHORT = "ReadRecords";
+    private static final String READRECORDSSHORT_OPERATION_IDSHORT = "ReadRecordsShort";
+
+    public enum Database {
+        influxDB,
+    }
+
```

Comment:
Two mandatory attributes along with their setters and getters are missing, Metadata and Segments.

Reply:
Not sure I understand the need for this. Could you please elaborate?

Reply:
Yes sure, as per the [specification document from IDTA](https://github.com/admin-shell-io/submodel-templates/tree/main/published/Time%20Series%20Data/1/1), there should be two attributes Metadata and Segments for a TimeSeries Submodel.

![image](https://github.com/eclipse-basyx/basyx-java-examples/assets/62088117/7f0600a2-af39-4e3c-8d34-a2825603d492)

![image](https://github.com/eclipse-basyx/basyx-java-examples/assets/62088117/523abc12-f4de-49a0-b140-125858256c9d)

Reply:
I see. The submodel class is only building the runtime operation, the rest of the spec I am taking directly from the aasx found in src/main/resources/admin-shells. I figured since those are all static information, the user can configure them in the aasx, and the java class will use the static information (metadata and segments) to build the operation properly. What do you think of this?

Reply:
I discussed this internally and this is fine with us. We can accept this remark as is.
Please resolve other remarks :)